src/Beacon.jsx

```diff
@@ -1,111 +1,206 @@
+          this.setState({ tooltipHidden: true });
+        }
+        , false);
+      }
+      // If the user specified an app root using the `TOOLTIP_OVERLAY_CLASS`
+      // then we fade out the background and highlight the target element of the tooltip.
+      if (tooltipActive) {
+        if (!oldClone) {
+          appRoot.className = `${TOOLTIP_FADED_CLASS} ${appRootClassName}`;
+          const targetClone = this.getTargetClone();
+          document.body.appendChild(targetClone);
+        }
+      } else {
+        appRoot.className = appRootClassName;
+      }
+    }
+
+    const tooltipClass = tooltipActive ? 'tour-in' : !this.state.tooltipHidden && 'tour-out';
+    return (
+      <TetherComponent
+        attachment={this.state.centerTooltip ? 'top center' : 'top left'}
+        constraints={[
+          {
+            to: 'scrollParent',
+            attachment: 'together'
+          }
+        ]}
+      >
+        {this.props.children}
+        <tour-tooltip class={tooltipClass} ref="tooltip" dangerouslySetInnerHTML={{__html: this.props.tooltipText}} />
+      </TetherComponent>
+    );
+  }
+
   render() {
-    const { parent, position, children } = this.props;
+    const { persistent } = this.state;
     if (!this.state.tooltip) {
-      if (!this.state.parentEl) {
-        return false;
-      }
-      const bounds = this.state.parentEl.getBoundingClientRect();
-      const { left, top } = this.getCoordinates(position, bounds);
-      const style = {
-        position: 'absolute',
-        left: `${left}px`,
-        top: `${top}px`
-      };
-      return (
-        <Portal isOpened>
-          <tour-beacon class="ignore-react-onclickoutside" style={style} onClick={::this.showTooltip} />
-        </Portal>
-      );
+      return this.renderBeacon(persistent);
     } else {
-      const style = {
-        style: { boxShadow: 'none' },
-        arrowStyle: {}
-      };
-      return (
-        <Portal isOpened>
-          <Tooltip
-            parent={parent}
-            tooltipTimeout={200}
-            style={style}
-            active={this.state.tooltipActive}
-            position={position}
-          >
-            {children}
-          </Tooltip>
-        </Portal>
-      );
+      return this.renderTooltip();
     }
   }
 
-  handleClickOutside() {
-    if (this.state.tooltip) {
+  handleClickOutside(event) {
+    if (this.state.tooltip && this.state.tooltipActive) {
+      event.preventDefault();
       this.setState({ tooltipActive: false });
     }
   }
 
-  showTooltip() {
+  showTooltip(event) {
+    event.preventDefault();
     this.setState({ tooltip: true });
+    if (this.state.hash) {
+      // Store the hash so we know this beacon has been clicked
+      const transaction = this.state.database.transaction(['beacons'], 'readwrite');
+      transaction.objectStore('beacons').add(true, this.state.hash);
+    }
   }
-});
+}
+
+export default withClickOutside(Beacon);
+export const BeaconConfig = Config;
```

Comment:
this is wrong... create `index.js` where you import both and then re-export them ... put the `index.js` as `main` in the `package.json`.